test/Microsoft.AspNet.Razor.Runtime.Test/DefaultTagHelperContentTest.cs

```diff
@@ -0,0 +1,201 @@
+
+        [Fact]
+        public void CanIdentifyIfModified()
+        {
+            // Arrange
+            var tagHelperContent1 = new DefaultTagHelperContent();
+            var tagHelperContent2 = new DefaultTagHelperContent();
+            var tagHelperContent3 = new DefaultTagHelperContent();
+
+            // Act
+            tagHelperContent1.SetContent("Test");
+            tagHelperContent2.Append("Foo");
+            tagHelperContent3.Clear();
+
+            // Assert
+            Assert.True(tagHelperContent1.IsModified);
+            Assert.True(tagHelperContent2.IsModified);
+            Assert.True(tagHelperContent3.IsModified);
+        }
+
+        [Fact]
+        public void CanAddWhiteSpaces()
+        {
+            // Arrange
+            var tagHelperContent = new DefaultTagHelperContent();
+
+            // Act
+            tagHelperContent.SetContent("  ");
+
+            // Assert
+            Assert.Equal(2, tagHelperContent.ToString().Length);
+        }
+
+        [Fact]
+        public void CanIdentifyWhiteSpaces()
+        {
+            // Arrange
+            var tagHelperContent = new DefaultTagHelperContent();
+
+            // Act
+            tagHelperContent.SetContent("  ");
+            tagHelperContent.Append(" ");
+
+            // Assert
+            Assert.True(tagHelperContent.IsWhiteSpace);
+        }
+
+        [Fact]
+        public void CanIdentifyIsEmpty()
+        {
+            // Arrange
+            var tagHelperContent = new DefaultTagHelperContent();
+
+            // Act & Assert
+            Assert.True(tagHelperContent.IsEmpty);
+        }
+
+        [Fact]
+        public void CanClearContent()
+        {
+            // Arrange
+            var tagHelperContent = new DefaultTagHelperContent();
+            tagHelperContent.SetContent("Hello World");
+
+            // Act
+            tagHelperContent.Clear();
+
+            // Assert
+            Assert.True(tagHelperContent.IsEmpty);
+        }
+
+        [Fact]
+        public void CanAppendWithoutSet()
+        {
+            // Arrange
+            var tagHelperContent = new DefaultTagHelperContent();
+            var expected = "Hello World!";
+
+            // Act
+            tagHelperContent.Append(expected);
+
+            // Assert
+            Assert.Equal(expected, tagHelperContent.GetContent());
+        }
+
+        [Theory]
+        [InlineData("HelloWorld!")]
+        [InlineData("  ")]
+        public void CopyToWorksAsExpected(string expected)
+        {
+            // Arrange
+            var tagHelperContent = new DefaultTagHelperContent();
+            var copiedTagHelperContent = new DefaultTagHelperContent();
+            tagHelperContent.SetContent(expected);
+
+            // Act
+            tagHelperContent.CopyTo(copiedTagHelperContent);
+
+            // Assert
+            Assert.Equal(expected, tagHelperContent.GetContent());
+        }
+
+        [Fact]
+        public void CanWriteToTextWriter()
```

Comment:
Ditto here, ensure that multiple appends all make their way into the text writer.